kubernetes/kiali_monitoring_types.go

```diff
@@ -19,7 +19,8 @@ var kialiMonitoringGroupVersion = schema.GroupVersion{
 }
 
 type MonitoringDashboard struct {
-	Spec MonitoringDashboardSpec
+	Metadata map[string]interface{}
```

Comment:
If you need it, you can map the Metadata with the Kubernetes Metadata type to get all the mappings done.
I think CRDs follow same type (if I'm now wrong here).
Probably you can save a couple of checks, but it's also ok to leave it as it is here.